### PR TITLE
wayland: set scale factor without wegl_surface (for WebGpu)

### PR DESCRIPTION
**Pull request by valpackett**
This fixes wrong scale with the WebGpu backend. Related: #2756

### CODE REVIEW

File: window/src/os/wayland/window.rs [hunk 1]
```diff
@@ -713,6 +713,9 @@ impl WaylandWindowInner {
                                 self.surface_factor = factor;
                             }
                         }
+                    } else if self.surface_factor != factor {
+                        // no wegl_surface for WebGpu, still have to set this
+                        self.surface.set_buffer_scale(factor);
```
Comment (wez) [inline window/src/os/wayland/window.rs:718]: I think the same buffer replacement stuff that is done above is likely also needed here, as well as setting `self.surface_factor = factor` for consistency.

Perhaps the logic should be something like this:

```rust
                        if let Some(wegl_surface) = self.wegl_surface.as_mut() {
                             wegl_surface.resize(pixel_width, pixel_height, 0, 0);
                        }
                        
                        if self.surface_factor != factor {
                            let wayland_conn = Connection::get().unwrap().wayland();
                            let mut pool = wayland_conn.mem_pool.borrow_mut();
                            // Make a "fake" buffer with the right dimensions, as
                            // simply detaching the buffer can cause wlroots-derived
                            // compositors consider the window to be unconfigured.
                            if let Ok((_bytes, buffer)) = pool.buffer(
                                factor,
                                factor,
                                factor * 4,
                                wayland_client::protocol::wl_shm::Format::Argb8888,
                            ) {
                                self.surface.attach(Some(&buffer), 0, 0);
                                self.surface.set_buffer_scale(factor);
                                self.surface_factor = factor;
                            }
                        }
```

can you try that out both with and without webgpu?

Comment (valpackett) [reply window/src/os/wayland/window.rs]: Makes sense. Though I just started testing on wlroots hidpi and actually nothing works, everything I've tried results in wezterm getting disconnected due to an `error in client communication`. :/

Comment (wez) [reply window/src/os/wayland/window.rs]: This fell off my radar over the holidays, just catching back up now.
Is this PR good to merge? It wasn't clear whether you had issues related to this or not

Comment (valpackett) [reply window/src/os/wayland/window.rs]: I think it's an improvement either way as it works on gnome, but I haven't managed to fix the error on wlroots :/

Comment (wez) [reply window/src/os/wayland/window.rs]: Ok, thanks!
For debugging your wlroots issue, have you tried running with WAYLAND_DEBUG=1 set in the environment to see if that reveals anything?

Comment (valpackett) [reply window/src/os/wayland/window.rs]: Of course! that only explains *when* something went wrong, not what. IIRC wlroots didn't log anything, so the disconnect was entirely unexplained. I'll need to come back to this later if no one else does..

**Pull request merged**